super-csv/src/main/java/org/supercsv/io/Tokenizer.java

```diff
@@ -228,31 +231,65 @@ else if( c == quoteChar ) {
 					potentialSpaces = 0;
 					currentColumn.append(c);
 				}
-				
+
 			} else {
-				
+
 				/*
 				 * QUOTE_MODE (within quotes).
 				 */
-				
-				if( c == quoteChar ) {
+
+				if( c == quoteEscapeChar ) {
 					int nextCharIndex = charIndex + 1;
 					boolean availableCharacters = nextCharIndex < line.length();
 					boolean nextCharIsQuote = availableCharacters && line.charAt(nextCharIndex) == quoteChar;
+					boolean nextCharIsEscapeQuoteChar = availableCharacters && line.charAt(nextCharIndex) == quoteEscapeChar;
+
 					if( nextCharIsQuote ) {
 						/*
-						 * An escaped quote (""). Add a single quote, then move the cursor so the next iteration of the
-						 * loop will read the character following the escaped quote.
+						 * An escaped quote (e.g. "" or \"). Skip over the escape char, and add
+						 * the following quote char as part of the column;
+						 */
+						charIndex++;
+						currentColumn.append(line.charAt(nextCharIndex));
+					} else if( nextCharIsEscapeQuoteChar ) {
+						/*
+						 * A double escape (normally \\). Save the escape char, then continue to
+						 * next character.
 						 */
 						currentColumn.append(c);
 						charIndex++;
-						
-					} else {
+					} else if( quoteEscapeChar == quoteChar ) {
```

Comment:
@ash211  I prefer not - but hey I understand your reluctance. So lets be practical and save that refactoring for another day. May I then propose adding extra tests for the tokenization. The problem with the existing tests are that `"` and `\` cannot readily by inserted into java strings... hence they need escaping.. but we are already testing escaping..gaah. I propose copy the existing tests you've made and use `'` as quote and `$` as escape or something..  also this test file is growing out of hand - maybe extract all your introduced tests into a fresh file?